src/auth/Auth.sol

```diff
@@ -42,14 +42,63 @@ abstract contract Auth {
         emit AuthorityUpdated(authority);
     }
 
-    function isAuthorized(address user, bytes4 functionSig) internal view virtual returns (bool) {
-        Authority cachedAuthority = authority;
-
-        if (address(cachedAuthority) != address(0)) {
-            if (cachedAuthority.canCall(user, address(this), functionSig)) return true;
+    function isAuthorized(address user, bytes4 functionSig) internal view virtual returns (bool authorized) {
+        assembly {
+            let cachedAuthority := sload(authority.slot)
+
+            if iszero(eq(cachedAuthority, 0)) {
+                // We'll use 4 + 32 * 3 bytes.
+                let callDataLength := 100
+
+                // Get a pointer to some free memory.
+                let freeMemoryPointer := mload(0x40)
+
+                // Update the free memory pointer for safety.
+                mstore(0x40, add(freeMemoryPointer, callDataLength))
```

Comment:
Do we need to do this? Won't the compiler safely clean before writing to the slot again?